staging/src/k8s.io/component-base/logs/json/json.go

```diff
@@ -81,9 +81,13 @@ func epochMillisTimeEncoder(_ time.Time, enc zapcore.PrimitiveArrayEncoder) {
 // Factory produces JSON logger instances.
 type Factory struct{}
 
-var _ registry.LogFormatFactory = Factory{}
+var _ logsapi.LogFormatFactory = Factory{}
 
-func (f Factory) Create(options config.FormatOptions) (logr.Logger, func()) {
+func (f Factory) Feature() featuregate.Feature {
+	return logsapi.LoggingBetaOptions
```

Comment:
I should point out that this will break current users of invocations like `kube-scheduler --logging-format=json`: they need to add `--feature-gate LoggingBetaOptions=true`.

Perhaps it would be better to be a bit less precise and not mark JSON output as beta here?

Reply:
Scratch that. The beta feature is enabled by default and the alpha features are something where breaking changes are okay.